datafusion/common/src/dfschema.rs

```diff
@@ -729,6 +729,34 @@ impl From<Field> for DFField {
     }
 }
 
+/// DataFusion-specific extensions to [`Schema`].
+pub trait SchemaExt {
+    /// This is a specialized version of Eq that ignores differences
+    /// in nullability and metadata.
+    ///
+    /// It works the same as [`DFSchema::equivalent_names_and_types`].
+    fn equivalent_names_and_types(&self, other: &Self) -> bool;
+}
+
+impl SchemaExt for Schema {
+    fn equivalent_names_and_types(&self, other: &Self) -> bool {
+        if self.fields().len() != other.fields().len() {
+            return false;
+        }
+
+        self.fields()
+            .iter()
+            .zip(other.fields().iter())
+            .all(|(f1, f2)| {
+                f1.name() == f2.name()
+                    && DFSchema::datatype_is_semantically_equal(
```

Comment:
since we use `DFSchema` reference anyway, prob we can reuse `DFSchema::equivalent_names_and_types` without introducing new method. 🤔 Moreover we have Schema <-> DFSchema converters in place.

Otherwise we have to test this method thoroughly

Reply:
I think this might be more convenient to use 🤔 .
All the test cases from DFSchema have been reused for testing.

Reply:
I think the problem is that `DFSchema` isn't a wrapper on top of `Schema` (so you can't get a `&Schema` from a `&DFSchema`)

@jonahgao  has updated the tests to cover this new code, so I think we are good